src/backend/core/warren/base_indicator.py

```diff
@@ -2,9 +2,75 @@
 from abc import ABC, abstractmethod
 from typing import List
 
+import numpy as np
+import pandas as pd
 from ralph.backends.http.lrs import LRSQuery
 
 
+def parse_raw_statements(raw_statements) -> pd.DataFrame:
+    """Parse LRS statements, explode the columns, and add a `date` column."""
+    flattened = pd.json_normalize(raw_statements)
+    # Disable chained assignment warning to make the transformation inplace
+    pd.options.mode.chained_assignment = None
+    # Transform timestamp column into a date with day (YYYY-MM-DD)
+    flattened.loc[:, "date"] = pd.to_datetime(flattened.loc[:, "timestamp"]).dt.date
+    return flattened
+
+
+def add_uniformized_actor_id(statements: pd.DataFrame) -> pd.DataFrame:
```

Comment:
I would prefer the following:

```suggestion
def add_actor_unique_id(statements: pd.DataFrame) -> pd.DataFrame:
```